guides/resource-provisioning/postgres-database.md

```diff
@@ -0,0 +1,189 @@
+# Postgres Database
+
+This guide describes how to add a Postgres database to a microservice running on the FFC Platform Kubernetes cluster.
+
+## Create Managed Identity for your microservice
+
+If not alreay configured add [Managed Identity](managed-identity.md) to your microservice
+
+## Request creation of microservice database
+
+Request Cloud Services through your usual support channel to create a database within your Azure Database for PostgreSQL following the naming convention:
+
+```
+ffc_<workstream>_<service>
+```
+
+for example `ffc_demo_claim`.
+
+*Note here the use of underscores instead of the normal hyphen convention. Postgres hyphens require escaping with double quote marks so underscores are preferred.*
+
+## Request creation of microservice database role
+
+Request Cloud Services to create a database role that is bound to the Managed Identity created for the microservice ([Azure guidence](https://docs.microsoft.com/en-us/azure/postgresql/howto-connect-with-managed-identity#creating-a-postgresql-user-for-your-managed-identity)), for example `ffc-snd-demo-claim-role`.
+
+## Create a Liquibase changelog
+
+The FFC Platfrom CI and deployment pipelines support database migrations using [Liquibase](https://www.liquibase.org/).
+
+Create a Liquibase changelog defining the structure of your database available from the root of your microservice repoository in `changelog/db.changelog.xml`.
+
+Guidence on creating a Liquibase changelog is outside of the scope of this guide, so please check current best practice with the FFC Platform Team.
+
+## Update Docker Compose files to use Postgres service and environment variables
+
+Add the following configuration to the microservice `docker-compose.yaml`:
+
+```
+    depends_on:
+      - ffc-<workstream>-<service>-postgres
+    environment:
+      POSTGRES_DB: ffc_<workstream>_<service>
+      POSTGRES_PASSWORD: ppp
+      POSTGRES_USER: postgres
+      POSTGRES_HOST: ffc-<workstream>-<service>-postgres
+      POSTGRES_PORT: 5432
+      POSTGRES_SCHEMA_NAME: public
+
+  ffc-<workstream>-<service>-postgres:
+    image: postgres:11.4-alpine
+    environment:
+      POSTGRES_DB: ffc_<workstream>_<service>
+      POSTGRES_PASSWORD: ppp
+      POSTGRES_USER: postgres
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+
+volumes:
+  postgres_data: {}
```

Comment:
John opinion time.  I don't like the postgres definition being in the main compose file.  I think the main compose file should only include the minimum to run the service and other files should add what is needed for local development and test outside of building that final image.

I think that makes services more predictable.

Now that does mean that you end up duplicating the postgres definition in the `test` and `override` files, but personally I see that as a good thing as you can isolate test data from local development data.  Here's an example of how I tend to do it -> https://github.com/johnwatson484/dream-league-api


Anyway, I guess the point I'm making is that different development teams will have different ideas over the optimal setup and I don't want to be too strong about saying they must put the postgres definition in that particular file, especially as I personally disagree with it.  So perhaps we could change the wording to say it needs to be in the compose files, so it can be used locally and in CI.  For example, the simplest solution is to add it to `docker-compose.yaml`.

Reply:
Agree. Language updated to be a bit more general